packages/suite/src/storage/migrations/index.ts

```diff
@@ -890,4 +890,207 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
+            ) {
+                suiteSettings.evmSettings.confirmExplanationModalClosed.pol =
+                    // @ts-expect-error
+                    suiteSettings.evmSettings.confirmExplanationModalClosed.matic;
+                // @ts-expect-error
+                delete suiteSettings.evmSettings.confirmExplanationModalClosed.matic;
+            }
+
+            if (
+                // @ts-expect-error
+                typeof suiteSettings.evmSettings.explanationBannerClosed?.matic == 'boolean'
+            ) {
+                suiteSettings.evmSettings.explanationBannerClosed.pol =
+                    // @ts-expect-error
+                    suiteSettings.evmSettings.explanationBannerClosed.matic;
+                // @ts-expect-error
+                delete suiteSettings.evmSettings.explanationBannerClosed.matic;
+            }
+
+            return suiteSettings;
+        });
+
+        const backendSettings = transaction.objectStore('backendSettings');
+        // @ts-expect-error
+        const maticBackendSettings = await backendSettings.get('matic');
+        if (maticBackendSettings) {
+            backendSettings.add(maticBackendSettings, 'pol');
+            // @ts-expect-error
+            backendSettings.delete('matic');
+        }
+
+        const tokenManagement = transaction.objectStore('tokenManagement');
+        const maticTokenManagementShow = await tokenManagement.get('matic-coin-show');
+        if (maticTokenManagementShow) {
+            tokenManagement.add(maticTokenManagementShow, 'pol-coin-show');
+            tokenManagement.delete('matic-coin-show');
+        }
+
+        const maticTokenManagementHide = await tokenManagement.get('matic-coin-hide');
+        if (maticTokenManagementHide) {
+            tokenManagement.add(maticTokenManagementHide, 'pol-coin-hide');
+            tokenManagement.delete('matic-coin-hide');
+        }
+
+        await updateAll(transaction, 'discovery', discovery => {
+            discovery.networks = discovery.networks.filter(
+                // @ts-expect-error
+                network => network !== 'matic',
+            );
+            discovery.networks = discovery.networks.map(network =>
+                // @ts-expect-error
+                network === 'matic' ? 'pol' : network,
+            );
+
+            discovery.failed = discovery.failed.map(network => {
+                // @ts-expect-error
+                if (network.symbol === 'matic') {
+                    network = { ...network, symbol: 'pol' };
+                }
+
+                return network;
+            });
+
+            return discovery;
+        });
+
+        await updateAll(transaction, 'accounts', account => {
+            // @ts-expect-error
+            if (account.symbol === 'matic') {
+                account = { ...account, symbol: 'pol' };
+            }
+
+            return account;
+        });
```

Comment:
the migration stops here and does not continue, any idea why? @marekrjpolak

Reply:
![Screenshot 2024-09-26 at 21 57 11](https://github.com/user-attachments/assets/0ee3f24d-16e9-4387-8d53-7a2686c3c3cd)

Reply:
Resolved in person. Long story short, `updateAll` util is not capable of updating entries including their primary key, so they have to be removed and re-added instead.